app/code/Magento/Catalog/Model/Product/Authorization.php

```diff
@@ -32,14 +34,24 @@ class Authorization
      */
     private $productFactory;
 
+    /**
+     * @var RequestInterface
+     */
+    private $request;
+
     /**
      * @param AuthorizationInterface $authorization
      * @param ProductFactory $factory
+     * @param RequestInterface|null $request
      */
-    public function __construct(AuthorizationInterface $authorization, ProductFactory $factory)
-    {
+    public function __construct(
+        AuthorizationInterface $authorization,
+        ProductFactory $factory,
+        ?RequestInterface $request = null
```

Comment:
Dont think you should use Request inside model.
Please change implementation to not involve request to model.